lib/webidl2.js

```diff
@@ -370,16 +358,12 @@
         if (args = argument_list(store)) {
           ret["arguments"] = args;
         }
-        // [MapClass(DOMString, DOMString)]
-        else if (pair = type_pair()) {
-          ret.typePair = pair;
-        }
         // [Constructor()]
         else {
           ret["arguments"] = [];
         }
         all_ws();
-        consume(OTHER, ")") || error("Unexpected token in extended attribute argument list or type pair");
+        consume(OTHER, ")") || error("Unexpected token in extended attribute argument list");
```

Comment:
Can you include the erroneous token in the error? `": " + token)`. Makes it easier when reporting issues.